test/jit/test_cuda.py

```diff
@@ -445,16 +445,54 @@ def test_wait_event():
             return e0.query() and s0.query() and s1.query()
         self.assertTrue(test_wait_event())
 
+        # Test asynchronous task operations enqueued on multiple CUDA Streams.
+        def test_multi_stream_async_ops():
```

Comment:
I think I commented on this in the original streams PR but it's weird to have all of these tests in one method of the `TestCUDA` class. I would split these up as much as possible so that failures are easier to debug.